packages/consent-manager/index.tsx

```diff
@@ -0,0 +1,123 @@
+  const [showBanner, setShowBanner] = useState(() => {
+    // 1. If prop override is set, always show the consent bar.
+    if (props.forceShow) return true
+  })
+
+  const hasEmptyPreferencesOrVersionMismatch =
+    Object.keys(preferences).length === 0 ||
+    preferences.version !== props.version
+
+  const saveAndLoadAnalytics = (preferences) => {
+    if (typeof preferences === 'undefined') {
+      preferences = { loadAll: false, segment: false }
+    }
+    savePreferences(preferences, props.version)
+
+    // If analytics have already been added to page, it's likely you're updating your preferences
+    // We reload the page to re-initiate the script with the updated integrations
+    // @ts-expect-error -- initialized doesn't exist on the segment type?
+    if (window.analytics && window.analytics.initialized) {
+      window.location.reload()
+      return
+    }
+
+    // Close all dialogs
+    document.body.classList.remove('g-noscroll')
+    setShowDialog(false)
+    setShowBanner(false)
+    setPreferences(loadPreferences() ?? {})
+  }
+
+  const openDialog = useCallback(() => {
+    document.body.classList.add('g-noscroll')
+    setShowBanner(false)
+    setShowDialog(true)
+  }, [])
+
+  // Setup the event handler to open the dialog imperatively
+  useEffect(() => {
+    emitter.on('openDialog', openDialog)
+    return () => {
+      emitter.off('openDialog', openDialog)
+    }
+  }, [openDialog])
+
+  // Show banner if there are no preferences or the version mismatches
+  useEffect(() => {
+    setShowBanner(hasEmptyPreferencesOrVersionMismatch)
+    // eslint-disable-next-line react-hooks/exhaustive-deps -- We only want to run this check once
+  }, [])
+
+  return (
+    <div className={classNames(s.root, props.className)}>
+      {/*  Consent banner at the bottom */}
+      {showBanner && (
+        <ConsentBanner
+          privacyPolicyLink={props.privacyPolicyLink}
+          cookiePolicyLink={props.cookiePolicyLink}
+          onManagePreferences={openDialog}
+          onAccept={() => {
+            saveAndLoadAnalytics({ loadAll: true })
+          }}
+        />
+      )}
+      {/*  Consent manager preferences dialog */}
+      {showDialog && (
+        <ConsentPreferences
+          version={props.version}
+          segmentWriteKey={props.segmentWriteKey}
+          utilServerRoot={props.utilServerRoot}
+          segmentServices={props.segmentServices}
+          additionalServices={props.additionalServices}
+          preferences={preferences}
+          categories={props.categories}
+          privacyPolicyLink={props.privacyPolicyLink}
+          cookiePolicyLink={props.cookiePolicyLink}
+          saveAndLoadAnalytics={saveAndLoadAnalytics}
+        />
+      )}
+      <SegmentScript
+        preferences={preferences}
+        writeKey={props.segmentWriteKey}
+      />
+      <CustomScripts
+        preferences={preferences}
+        services={props.additionalServices}
+      />
```

Comment:
This replaces the previous `loadAnalytics` method. It's a bit more declarative and easier to trace now, these components will inject the necessary scripts based on the user's preferences.